### ISSUE TITLE
[BUG] Unable to calculate a request signature with connector ElasticSearch

### ISSUE THREAD

**Issue by BhuviTheDataGuy** [awslabs/aws-athena-query-federation#510]
Connector deployed and able to list the databases from Athena console, but when I select the list of tables, then Im getting this error.

```java
Error processing request to list indices: Unable to calculate a request signature: null: java.lang.RuntimeException
java.lang.RuntimeException: Error processing request to list indices: Unable to calculate a request signature: null
	at com.amazonaws.athena.connectors.elasticsearch.ElasticsearchMetadataHandler.doListTables(ElasticsearchMetadataHandler.java:211)
	at com.amazonaws.athena.connector.lambda.handlers.MetadataHandler.doHandleRequest(MetadataHandler.java:250)
	at com.amazonaws.athena.connector.lambda.handlers.CompositeHandler.handleRequest(CompositeHandler.java:135)
	at com.amazonaws.athena.connector.lambda.handlers.CompositeHandler.handleRequest(CompositeHandler.java:103)
Caused by: java.lang.RuntimeException: Unable to calculate a request signature: null
	at org.elasticsearch.client.RestClient.extractAndWrapCause(RestClient.java:812)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:225)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:212)
	at org.elasticsearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1433)
	at org.elasticsearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:1403)
	at org.elasticsearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:1373)
	at org.elasticsearch.client.IndicesClient.getAlias(IndicesClient.java:986)
	at com.amazonaws.athena.connectors.elasticsearch.AwsRestHighLevelClient.getAliases(AwsRestHighLevelClient.java:82)
	at com.amazonaws.athena.connectors.elasticsearch.ElasticsearchMetadataHandler.doListTables(ElasticsearchMetadataHandler.java:196)
	... 3 more
Caused by: com.amazonaws.SdkClientException: Unable to calculate a request signature: null
	at com.amazonaws.auth.AbstractAWSSigner.sign(AbstractAWSSigner.java:109)
	at com.amazonaws.auth.AWS4Signer.newSigningKey(AWS4Signer.java:641)
	at com.amazonaws.auth.AWS4Signer.deriveSigningKey(AWS4Signer.java:404)
	at com.amazonaws.auth.AWS4Signer.sign(AWS4Signer.java:253)
	at com.amazonaws.athena.connectors.elasticsearch.AWSRequestSigningApacheInterceptor.process(AWSRequestSigningApacheInterceptor.java:128)
	at org.apache.http.protocol.ImmutableHttpProcessor.process(ImmutableHttpProcessor.java:133)
	at org.apache.http.impl.nio.client.MainClientExec.prepareRequest(MainClientExec.java:520)
	at org.apache.http.impl.nio.client.MainClientExec.prepare(MainClientExec.java:146)
	at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.start(DefaultClientExchangeHandlerImpl.java:128)
	at org.apache.http.impl.nio.client.InternalHttpAsyncClient.execute(InternalHttpAsyncClient.java:141)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:221)
	... 10 more
Caused by: java.lang.NullPointerException
	at com.amazonaws.auth.AbstractAWSSigner.sign(AbstractAWSSigner.java:104)
	... 20 more
```

ES is in VPC, Connection is going via SSL, ES version Elasticsearch 7.9

**Comment by stefan-novak-brt** [comment]
I've come across the same issue and am curious if anyone has any new insights in the last six month since this issue was created.

Thanks!

**Comment by ykrystal** [comment]
We made a recent change that we believe should resolve this issue. For anybody who is still experiencing this, can you please build the connector to verify if the latest changes resolve your issue? Thank you!

**Comment by anelmu** [comment]
I just tried today, and have the same problem. Seems that issue is still present.

**Comment by roey-navina** [comment]
Same issue here

**Comment by dkefal** [comment]
We were able to get the Athena OpenSearch connector working with a workaround. 
The workaround involved setting the AutoDiscoverEndpoint to false and creating a VPC OpenSearch endpoint with the domain set to: domain=https://USERNAME@PASSWORD:VPC_ENDPOINT_DOMAIN/

While this workaround was successful which was probably bypassing the signature and resolving the issue, it has now broken due to the latest Opensearch software update R20221114-P4, which was hotfixed on March 17, 2023.

We would like to know if there is an estimated timeline for when this issue will be resolved.

**Comment by akuzin1** [comment]
Thank you for your patience. The [fix](https://github.com/awslabs/aws-athena-query-federation/pull/1157) should have been included in version 2023.13.1 of the Elasticsearch connector, so please upgrade to latest version and issue should be resolved. If any other questions arise, please reopen ticket.
